airflow/providers/google/cloud/hooks/dataplex.py

```diff
@@ -0,0 +1,228 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+"""This module contains Google Dataplex hook."""
+import os
+from time import sleep
+from typing import Any, Dict, Optional, Sequence, Union
+
+from google.api_core.retry import exponential_sleep_generator
+from googleapiclient.discovery import Resource, build
```

Comment:
not an expert on gcp but didn't we migrated out from discovery https://github.com/apache/airflow/pull/18184/files ?

Reply:
Yep. Python SDK is preferred indeed. Is there a problem with using it in this case @wojsamjan ?

Reply:
@eladkal @potiuk +1 for using dedicated SDK but if I correctly remember not all services have dedicated one.

Reply:
As @turbaszek mentioned, there is no SDK for Dataplex